### PR TITLE
Add types around workspace folders support

### PR DESCRIPTION
**Pull request by artempyanykh**
The main parts are:
1. Having WorkspaceFolders in InitializeParams as it supersedes rootPath and rootUri.
2. Having workspace folders server capability.

I've tested this in my server, and it seems to work.

### CODE REVIEW

**Comment by baronfel** [comment]
LGTM, thanks for contributing the types for this!

File: src/LanguageServerProtocol.fs [hunk 1]
```diff
@@ -848,14 +848,36 @@ module Types =
...
+    type ClientInfo = {
+        Name: string
+        Version: string option
+    }
 
     type InitializeParams = {
         ProcessId: int option
+        /// Information about the client.
+        /// @since 3.15.0
+        ClientInfo: ClientInfo option
```
Comment (baronfel) [inline src/LanguageServerProtocol.fs:870]: I don't know why, but CI thinks this field is duplicated on this structure.

Comment (artempyanykh) [reply src/LanguageServerProtocol.fs:870]: Hm. Maybe that's because my branch is forked off 0.3.1. Let me rebase on master and try again.

Comment (artempyanykh) [reply src/LanguageServerProtocol.fs:870]: Yeah, so between 0.3.1 and current main `ClientInfo` was added as a `JToken option`. I've removed the dupe favouring a properly typed version.

**Comment by baronfel** [comment]
I marked this to merge when the checks complete, and I'll be pushing out a new version of this repo hopefully this week so you don't have to submodule anymore.

**Comment by baronfel** [comment]
0.4.0 has been released! go forth and update!

**Pull request merged**